src/caf/toolkit/io.py

```diff
@@ -303,6 +304,84 @@ def _normalise_name(col: str) -> str:
     return re.sub(rf"[^{NORMALISED_CHARACTERS}]", "", normalised)
 
 
+def read_df(
+    path: os.PathLike,
+    index_col: list[int] | int | None = None,
+    find_similar: bool = False,
+    **kwargs,
+) -> pd.DataFrame:
+    """
+    Reads in the dataframe at path. Decompresses the df if needed.
+
+    Parameters
+    ----------
+    path:
+        The full path to the dataframe to read in
+
+    index_col:
+        Will set this column as the index if reading from a compressed
+        file, and the index is not already set.
+        If reading from a csv, this is passed straight to pd.read_csv()
+
+    find_similar:
+        If True and the given file at path cannot be found, files with the
+        same name but different extensions will be looked for and read in
+        instead. Will check for: '.csv', '.pbz2'
+
+    Returns
+    -------
+    df:
+        The read in df at path.
+    """
+
+    # Try and find similar files if we are allowed
+    if not os.path.exists(path):
+        if not find_similar:
+            raise FileNotFoundError(f"No such file or directory: '{path}'")
+        path = find_filename(path)
+
+    if pathlib.Path(path).suffix == ".csv":
+        return pd.read_csv(path, index_col=index_col, **kwargs)
+
+    if pathlib.Path(path).suffix in PD_COMPRESSION:
+        return pd.read_csv(path, index_col=index_col, **kwargs)
+
+    raise ValueError("Cannot determine the filetype of the given path.")
```

Comment:
I think this can be simplified. We either find a valid path in the `find_filename` that pandas can open, or we don't and the `find_filename` raises an error

```suggestion
    alt_suffix = list(PD_COMPRESSION) + [".csv"]

    # Try and find similar files if we are allowed
    if not os.path.exists(path):
        if not find_similar:
            raise FileNotFoundError(f"No such file or directory: '{path}'")
        path = find_similar_filename(path, alt_types=alt_suffix)

    if pathlib.Path(path).suffix in alt_suffix:
        return pd.read_csv(path, index_col=index_col, **kwargs)

    raise RuntimeError("A valid file was found, but apparently we can't load it??")
```